core/templates/dev/head/services/StateTopAnswersStatsService.js

```diff
@@ -50,24 +52,53 @@ oppia.factory('StateTopAnswersStatsService', [
       });
     };
 
+    $rootScope.$on('addState', function(event, args) {
+      if (stateTopAnswerStatsCache.hasOwnProperty(args.state_name)) {
+        // No action required, we already have data for that state.
+        return;
+      } else {
+        // Give the cache an empty list of answers.
+        stateTopAnswerStatsCache[args.state_name] = [];
+      }
+    });
+
+    $rootScope.$on('deleteState', function(event, args) {
+      // Take no action, we'll hold onto the stats in the cache.
```

Comment:
Why? I think we should delete it. Discarding a changelist is the only way to undo this and that forces a page reload.

Reply:
Done.